packages/jest-matchers/src/matchers.js

```diff
@@ -109,6 +109,35 @@ const matchers: MatchersObject = {
         `'${expected}' (using <=)`;
     return {message, pass};
   },
+
+  toBeCloseTo(actual: number, expected: number, precision: number) {
+    ensureNumbers(actual, expected, '.toBeCloseTo');
+    if (precision !== 0) {
+      precision = precision || 2;
+    }
+    const pass = Math.abs(expected - actual) < (Math.pow(10, -precision) / 2);
+    const message = pass
+      ? () => `expected '${actual}' not to be close to '${expected}'`
+      : () => `expected '${actual}' to be close to '${expected}'`;
+
+    return {message, pass};
+  },
+
+  toMatch(actual: any, expected: string) {
```

Comment:
i would throw in the case if `typeof actual !== 'string'` too.
Sometimes people do
`expect(returnStringOrUndefined()).toMatch(/lol/)` and it's pretty confusing to debug